fhir_extractor/src/app.py

```diff
@@ -30,43 +32,44 @@
 producer = ExtractorProducer(broker=os.getenv("KAFKA_BOOTSTRAP_SERVERS"))
 
 
-@app.route("/extractor_sql/<resource_id>/<primary_key_value>", methods=["POST"])
-def extractor_sql_single(resource_id, primary_key_value):
+@app.route("/extractor_sql", methods=["POST"])
+def extractor_sql_batch():
     """
-    Extract record for the specified resource and primary key value
-    :param resource_id:
-    :param primary_key_value:
-    :return:
+    Extract all records for the specified resource
     """
+    body = request.get_json()
+    resource_ids = body.get("resourceIds", None)
+
     try:
-        # TODO this routes only makes sense with a single resource mapping.
-        # This is mocked here by taking only the first resource mapping of the file.
-        # We should have a get_resource_mapping_by_id or something like that using resource_id.
-        resource_mapping = get_mapping(from_file="mapping_files/patient_mapping.json")[0]
+        # Get the resources we want to process from the pyrog mapping for a given source
+        resources = get_mapping(
+            resource_ids=resource_ids, from_file="mapping_files/patient_mapping.json"
+        )
 
-        analysis = analyzer.analyze(resource_mapping)
+        for resource_mapping in resources:
+            analysis = analyzer.analyze(resource_mapping)
 
-        run_resource(resource_mapping, analysis, [primary_key_value])
+            run_resource(resource_mapping, analysis)
 
         return "Success", 200
 
     except Exception as e:
         raise OperationOutcome(e)
 
 
-@app.route("/extractor_sql", methods=["POST"])
-def extractor_sql_batch():
+@app.route("/extractor_sql/<resource_id>/<primary_key_value>", methods=["POST"])
```

Comment:
I am not convinced by the structure of the 2 endpoints. Once, the resource_id is in the path, and the other time, it is in the body of the request. I think we should align both on that (body for both). WDYT ?

Reply:
I agree that it's a bit inconsistent.
How would you see it? 2 routes /batch and /single-row (or something else) and everything else in the query body?

Reply:
Yep, I think that is clearer